cocos/ui/UIPageView.cpp

```diff
@@ -513,28 +610,46 @@ void PageView::handleReleaseLogic(Touch *touch)
     {
         Vec2 curPagePos = curPage->getPosition();
         ssize_t pageCount = this->getPageCount();
-        float curPageLocation = curPagePos.x;
-        float pageWidth = getContentSize().width;
-        if (!_usingCustomScrollThreshold) {
-            _customScrollThreshold = pageWidth / 2.0;
+        
+        auto contentSize = getContentSize();
+       
+        float moveBoundray = 0.0f;
+        float scrollDistance;
+        if (_direction == Direction::HORIZONTAL)
```

Comment:
This will cause unwanted effect when user touch move pages in two direction. Because it adjust only one direction, not the other